### ISSUE TITLE
changes to page only rendered upon restarting logseq?

### ISSUE THREAD

**Issue by jaker-dotcom** [vipzhicheng/logseq-plugin-mark-map#82]
Hello there! 
Thanks for the plugin, I think it is promising for presenting stuff. However I found one critical thing not working and one big nuisance  on my end:
1. page links do work, but somehow the shortcut meta+[ doesn't take me back to the page before
2. when I edit the page I have to close logseq and open it again for the markmap to refresh
3. some unexpected shortcut behaviour + I'd wish I could  change shortcuts 

In this issue I'll describe the second one.

Steps to reproduce:
1. open logseq
2. from journal go to a page (in my case "Econometrics")
4. From there to the page of a book (in my case "StockWatson")
5. enter markmap
6. close markmap, edit some stuff (in my case just move things around)
7. open markmap again

Result:
See the same markmap as before edit.

Workaround:
Close logseq in between.

Maybe a refresh button would do the trick?

**Comment by vipzhicheng** [comment]
I was on Mac. When I edit the content of a page, and then enter markmap, I can see the updated view. So what's your env? and is there any special edit?

**Comment by jaker-dotcom** [comment]
Nope. Even if I just add a block with "test", it does not render upon entering markmap again.
Pop!_OS 22.04 LTS

Logseq Plugins
![image](https://github.com/user-attachments/assets/0907174b-a160-42ab-ba86-311ff156e5d0)

Logseq Version 0.10.9

**Comment by polskiTran** [comment]
I have this same error as well. Opening mind map will be stuck on the same mind map even if I open other page or add new content
logseq 0.10.9

**Comment by jaker-dotcom** [comment]
Is it maybe in conjunction with #81? I think it only occured to me when I had visited a specific page. 
Unfortunately I totally forgot about creating the demo content needed to figure out what causes the issue and don't have the time rn.

**Comment by polskiTran** [comment]
Seems like this bug only happens on my most recent graph and not older one (which is weird, i dont really know how a logseq graph could be tied with versioning somehow to be strike this difference) but rolling package back to 0.4.1 fix it temporarily for me. However the render image function does not work anymore.

Edit: Forgot to mention iam on mac

**Comment by jaker-dotcom** [comment]
Maybe it's not the age of the graph but the content within? As in my case, _if I recall correctly_ it appeared to be triggered by the contents of the pages I visited. 
After a logseq restart, the Plugin would work on other pages within the same graph.

**Comment by RajibTheKing** [comment]
I’ve encountered an issue, which I’ve attached below:
![Image](https://github.com/user-attachments/assets/86db46e2-a218-419a-8d76-238e237e32af)


When I switch pages, it doesn't render the new page correctly. Instead, it continues rendering the previous page. Restarting Logseq resolves the issue temporarily.

I am using  **_Ubuntu 24.04.1 LTS_**

**Comment by vipzhicheng** [comment]
@RajibTheKing 

I've set up the Ubuntu environment and am using the latest versions of Logseq along with the Markmap plugin. However, I'm unable to reproduce the issue you reported. 

Upon reviewing my code, I noticed that I'm leveraging the Logseq core API to retrieve the current page for preparing the markmap data. My suspicion is that this might be an issue with the Logseq API itself, though I'm unable to delve deeper into its internals at this stage. 

As such, a quick fix might not be feasible, but I'll continue monitoring this closely and keep you updated on any developments.

**Comment by kdasilva-pro** [comment]
@vipzhicheng This occurs when there is a PDF file attached to a node.

As a workaround, I add the property `mark-map-display:: hidden`

**Comment by jaker-dotcom** [comment]
Woah that makes sense. At least there's a PDF attached to the page aswell where I encountered issues.
